code/mob/living/carbon/human/dummies.dm

```diff
@@ -0,0 +1,47 @@
+/mob/living/carbon/human/tdummy
+	real_name = "Target Dummy"
+	var/shutup = FALSE
+	var/stam_monitor
+
+	New()
+		. = ..()
+		src.stam_monitor = new /obj/machinery/maptext_monitor/stamina(src)
+		src.AddComponent(/datum/component/health_maptext)
+		src.ensure_speech_tree().AddSpeechModifier(SPEECH_MODIFIER_TEST_DUMMY)
+
+	disposing()
+		QDEL_NULL(src.stam_monitor)
+		. = ..()
+
+/mob/living/carbon/human/tdummy/security
```

Comment:
Target dummy variants are tricky because one of the main utilities of the original target dummy is the ability to go `spawn tdu` and have it immediately pop up. So if you're going to make variants of it they shouldn't inherit from it or share the `tdu` snippet anywhere in their path. To avoid the copy-paste this would cause you could either make a macro or a component that adds the standard target dummy setup to `New` and `disposing`